core/src/query.rs

```diff
@@ -169,10 +338,46 @@ impl MongoStatement for MongoQuery {
             _ => Error::QueryExecutionFailed(e),
         };
 
-        let _guard = connection.runtime.enter();
-        let cursor: Cursor<Document> = connection
-            .runtime
-            .block_on(async { aggregate.await.map_err(map_query_error) })?;
+        let cursor: Cursor<Document> = if let Some(c) = collection {
```

Comment:
No need for so much code duplication.
You can simply do
```
    let mut aggregate : Aggregate = if let Some(c) = collection {
                        let collection = db.collection::<Document>(c.as_str());
            collection.aggregate(pipeline).comment(stmt_id)
        }
        else {
            db.aggregate(pipeline).comment(stmt_id)
        };
```
And then keep the common code as before.

You can even take out the duplication of .comment and add it after the initial creation of the aggregate variable.

Reply:
I originally tried this multiple ways, but rust doesn't allow it. Here's a patch a ran with the code you're suggesting: https://spruce.mongodb.com/version/66f2fd62a11f4700074e2f7f/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

Reply:
ah yes, because Collection is defined in the block.
Just define it outside. I renamed collection -> collection_name to make it clearer that's what it is.
```
let collection;
let mut aggregate : Aggregate = if let Some(c_name) = collection_name {
    collection = db.collection::<Document>(c_name.as_str());
    collection.aggregate(pipeline).comment(stmt_id)
}
else {
    db.aggregate(pipeline).comment(stmt_id)
};
....
```

That compiles fine for me